src/CFamily/CMake/CompilationConfigProvider.cs

```diff
@@ -121,8 +121,13 @@ public CompilationDatabaseEntry GetConfig(string filePath)
             }
         }
 
-        private bool IsHeaderFile(string filePath) => 
-            Path.GetExtension(filePath).Equals(".h", StringComparison.OrdinalIgnoreCase);
+        private bool IsHeaderFile(string filePath)
```

Comment:
Why not add this method to the `CFamilyShared` class?
It would also be easier to test all of the valid extensions, casing etc.